### PR TITLE
V3.8.0 pipeline api document

### PR DESCRIPTION
**Pull request by star-e**
Add api documents.
Renamed `QueueHint` enumerations.

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cocos/rendering/custom/pipeline.ts [hunk 1]
```diff
@@ -40,32 +40,157 @@ import { AccessType, CopyPair, LightInfo, MovePair, QueueHint, ResolvePair, Reso
 import { RenderWindow } from '../../render-scene/core/render-window';
 import { Model } from '../../render-scene/scene';
 
+/**
+ * @internal
```
Comment (minggo) [inline cocos/rendering/custom/pipeline.ts]: internal -> engineInternal

File: cocos/rendering/custom/pipeline.ts [hunk 2]
```diff
@@ -40,32 +40,157 @@ import { AccessType, CopyPair, LightInfo, MovePair, QueueHint, ResolvePair, Reso
...
+ * It is used internally and should not be called directly.
+ * @zh PipelineRuntime是经典管线以及自定义管线的运行时。
+ * 属于内部实现，用户不应直接调用。
+ */
 export interface PipelineRuntime {
+    /**
+     * @en Activate PipelineRuntime with default swapchain
+     * @zh 用默认交换链初始化PipelineRuntime
+     */
     activate (swapchain: Swapchain): boolean;
```
Comment (minggo) [inline cocos/rendering/custom/pipeline.ts:57]: Missing docs for parameter and return value.

File: cocos/rendering/custom/pipeline.ts [hunk 3]
```diff
@@ -80,150 +205,544 @@ export function getPipelineTypeName (e: PipelineType): string {
...
+ */
 export interface RenderNode {
+    /**
+     * @en Get debug name of current node.
+     * @zh 获得当前节点调试用的名字
+     */
     name: string;
     /**
-     * @beta function signature might change
+     * @experimental
```
Comment (minggo) [inline cocos/rendering/custom/pipeline.ts:276]: I am not sure if our documentation tool supports it.

File: native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h [hunk 4]
```diff
@@ -77,6 +77,13 @@ namespace cc {
 
 namespace render {
 
+/**
+ * @internal
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h]: internal -> engineInternal

File: cocos/rendering/custom/pipeline.ts (hunk 1 above)
Comment (star-e) [reply cocos/rendering/custom/pipeline.ts]: I will fix all `engineInternal`s

File: native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h [hunk 5]
```diff
@@ -500,6 +1276,9 @@ class SceneVisitor {
     virtual void draw(const gfx::DrawInfo &info) = 0;
 };
 
+/**
+ * @deprecated @en Not used @zh 未使用
+ */
 class SceneTask {
```
Comment (minggo) [inline native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h:1299]: Can just remote it, as it is not used outside.

Comment (star-e) [reply native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h:1299]: Currently web-pipeline implementation relies on them. They will be removed in the future.

Comment (minggo) [reply native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h:1299]: Got it.

**Review by pandamicro — APPROVED** [review]

File: native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h (hunk 4 above)
Comment (star-e) [reply native/cocos/renderer/pipeline/custom/RenderInterfaceTypes.h]: all `@engineInternal` fixed

**Review by minggo — APPROVED** [review]

**Pull request merged**